pkg/ingester/flush.go

```diff
@@ -275,6 +287,11 @@ func (i *Ingester) flushUserSeries(flushQueueIndex int, userID string, fp model.
 
 	userState.fpLocker.Unlock(fp)
 
+	// No need to flush these chunks again.
+	for len(chunks) > 0 && chunks[0].flushed {
```

Comment:
Question: is it always guaranteed that chunks are flushed in order? May end up in a situation where X is not flushed but X+1 is?

Reply:
Not in the current code; `Put()` is always implemented by iterated calls to `PutOne()`, and any error considers the whole set to have failed.
But there's nothing guaranteeing this.  Possibly we should remove `Put()` to avoid confusion.